### PR TITLE
only return years greater than horizon

### PR DESCRIPTION
**Pull request by gidden**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Adding to RELEASE_NOTES.md

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

### CODE REVIEW

**Comment by gidden** [comment]
big questions about `astype(int)`. we currently dont do any typing on our time values, which.. yeah probably isn't good. at the moment `scen.set('year')` returns with `dtype == 'object'` and there are string/int issues in comparison..

**Comment by gidden** [comment]
ready for a look @danielhuppmann

File: message_ix/core.py [hunk 1]
```diff
@@ -109,12 +109,27 @@ def add_horizon(scenario, data):
...
     def vintage_and_active_years(self):
-        """Return a 2-tuple of valid pairs of vintage years and active years for
-        use with data input.
+        """Return a 2-tuple of valid pairs of vintage years and active years
+        for use with data input. A valid year-vintage, year-active pair is
+        one in which:
+
+        - year-vintage <= year-active
+        - both within the model's 'year' set
+        - year-active <= the model's first year
```
Comment (danielhuppmann) [inline message_ix/core.py]: this is `>=`, if I understand the code correctly...

File: message_ix/core.py [hunk 2]
```diff
@@ -109,12 +109,27 @@ def add_horizon(scenario, data):
...
         """
         horizon = self.set('year')
-        combinations = itertools.product(horizon, horizon)
-        year_pairs = [(y_v, y_a) for y_v, y_a in combinations if y_v <= y_a]
+        filters = {'type_year': ['firstmodelyear']}
+        first = (self
```
Comment (danielhuppmann) [inline message_ix/core.py]: can be written as `self.cat('year', 'firstmodelyear')[0] or self.set('year')[0]`

**Comment by danielhuppmann** [comment]
nice, @gidden!

**Comment by gidden** [comment]
wait on this

**Comment by gidden** [comment]
it touches the tutorials, so I want to put it on top of the hackathon PRs which @ClaraLuisa also touched

**Pull request closed without merging**